projects/plugins/jetpack/extensions/blocks/cookie-consent/cookie-consent.php

```diff
@@ -80,3 +82,14 @@ function load_assets( $attr, $content ) {
 	);
 }
 
+/**
+ * Since the cookie consent is part of the cached response HTML, it can still render even when the cookie is set (when it shouldn't).
+ * Because, by default, the cache doesn't vary around the cookie's value.
+ * This let's the cache know that the content has changed to return fresh content.
+ */
+function notify_batcache_that_content_changed() {
+	if ( function_exists( 'vary_cache_on_function' ) ) {
+		// Cast the cookie down to a boolean, to avoid arbitrary code execution.
```

Comment:
I don't understand that comment; am I missing something here?

Reply:
Yes, you can return the cookie as is as a cache key, `return $_COOKIE[ "' . COOKIE_NAME . '" ]`, but then anyone could run PHP code on our servers by populating the cookie with arbitrary code.

Reply:
> but then anyone could run PHP code on our servers by populating the cookie with arbitrary code.

I don't see any code path that would do that. The only use of the returned value that I see is that it is dumped into an array which is then run through `md5(serialize())`, and then that hash is used as a key for the cache entry.

OTOH, since all we care about is whether the cookie is set and not what its value is, it's probably best to do the same here to avoid unnecessary cache splitting.